application/src/main/java/org/togetherjava/tjbot/commands/github/GitHubReference.java

```diff
@@ -0,0 +1,169 @@
+package org.togetherjava.tjbot.commands.github;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import org.apache.commons.collections4.ListUtils;
+import org.kohsuke.github.*;
+import org.togetherjava.tjbot.annotations.MethodsReturnNonnullByDefault;
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.config.Config;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.awt.*;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.List;
+
+/**
+ * GitHub Referencing feature. If someone sends #id of an issue (e.g. #207) the bot replies with an
+ * embed that contains info on the issue/pr
+ */
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class GitHubReference extends MessageReceiverAdapter {
+    /**
+     * The pattern used to determine whether a message is referencing an issue
+     */
+    protected static final Pattern ISSUE_REFERENCE_PATTERN =
+            Pattern.compile("(?:\\s+|[()\\[\\]<>]|^)#(\\d+)(?:\\s+|[()\\[\\]<>]|$)");
```

Comment:
is there anyway this can become readable, like, groups with names or such?

Reply:
there's only one _real_ group. ?: indicates a noncapturing group

Reply:
Is there any way you could document this,. since I really don't understand what everything is exactly for.

I'd imagine all you need is a check which goes for: anycharacter#anynumbers

While here, it's a lot, don't know what it all does?

Reply:
I simplified the pattern